torch/testing/_internal/opinfo/definitions/signal.py

```diff
@@ -0,0 +1,275 @@
+
+    # Tests for negative decay values.
+    yield ErrorInput(
+        SampleInput(3, tau=-1, dtype=torch.float32, device=device, **kwargs),
+        error_type=ValueError,
+        error_regex="Tau must be positive, got: -1 instead.",
+    )
+
+    # Tests for non-symmetric windows and a given center value.
+    yield ErrorInput(
+        SampleInput(3, center=1, sym=False, dtype=torch.float32, device=device),
+        error_type=ValueError,
+        error_regex="Center must be 'None' for non-symmetric windows",
+    )
+
+
+def error_inputs_gaussian_window(op_info, device, **kwargs):
+    # Yield common error inputs
+    yield from error_inputs_window(
+        op_info,
+        device,
+        0.5,  # std
+        **kwargs
+    )
+
+    # Tests for negative standard deviations
+    yield ErrorInput(
+        SampleInput(3, -1, dtype=torch.float32, device=device, **kwargs),
+        error_type=ValueError,
+        error_regex="Standard deviation must be positive, got: -1 instead.",
+    )
+
+
+def make_signal_windows_opinfo(
+        name, variant_test_name, ref, sample_inputs_func, error_inputs_func, *, skips=()
+):
+    r"""Helper function to create OpInfo objects related to different windows."""
+    return OpInfo(
+        name=name,
+        variant_test_name=variant_test_name,
+        ref=ref if TEST_SCIPY else None,
+        dtypes=floating_types_and(torch.bfloat16, torch.float16),
+        dtypesIfCUDA=floating_types_and(torch.bfloat16, torch.float16),
+        sample_inputs_func=sample_inputs_func,
+        error_inputs_func=error_inputs_func,
+        supports_out=False,
+        supports_autograd=False,
+        skips=skips,
+    )
+
+
+op_db: List[OpInfo] = [
+    make_signal_windows_opinfo(
+        name="signal.windows.cosine",
+        variant_test_name="signal.windows.cosine_default",
+        ref=scipy.signal.windows.cosine,
+        sample_inputs_func=sample_inputs_window,
+        error_inputs_func=error_inputs_window,
+        skips=(
+            DecorateInfo(
+                unittest.expectedFailure,
+                "TestNormalizeOperators",
+                "test_normalize_operator_exhaustive",
+            ),
+            # TODO: same as this?
+            # https://github.com/pytorch/pytorch/issues/81774
+            # also see: arange, new_full
+            # fails to match any schemas despite working in the interpreter
+            DecorateInfo(
+                unittest.expectedFailure,
+                "TestOperatorSignatures",
+                "test_get_torch_func_signature_exhaustive",
+            ),
+            # fails to match any schemas despite working in the interpreter
+            DecorateInfo(
+                unittest.expectedFailure, "TestJit", "test_variant_consistency_jit"
+            ),
+            # skip these tests since we have non tensor input
+            DecorateInfo(
+                unittest.skip("Skipped!"), "TestCommon", "test_noncontiguous_samples"
+            ),
```

Comment:
Many of these can be moved to the `make_signal_windows_opinfo` function. See for example how  the OpInfo for `mvlgamma` does it.